upgrade/steps.go

```diff
@@ -836,6 +836,7 @@ func migrationSteps(ctx Context, repo ProviderRepo, providerName string, descrip
 
 func AddAutoAliasing(ctx Context, repo ProviderRepo, providerName string) (step.Step, error) {
 	steps := []step.Step{}
+	providerName = strings.TrimPrefix(providerName, "pulumi-")
```

Comment:
Removed in the previous refactor by accident

Reply:
This is also going to change line L847. Is that correct?

Also, we shouldn't do anything while producing a `step.Step`. We need to put L839-845 into a `step.F`.

Reply:
Yes